### PR TITLE
Multivariate linear regression

### PR DESCRIPTION
**Pull request by salvisolamartinell**
# Description

Implemented multivariate linear regression using ordinary least squares. Now, LinearRegression supports fitting a multivariate x (multiple explanatory variables (n_features>1)). The implementation is just linear algebra. Arrays of size n_features^2 are used.

## Type of change

Please delete options that are not relevant.

- [x] New algorithm or support class.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. If you haven't added any test and it is relevant provide instructions so we can reproduce.

- [ ] I have added a new test file: [E.g. `test_rf.py`]
- [x] I have added a new test case: [E.g. `RFTest.test_make_classification`]
- [x] I have tested it manually in a **local environment**.
- [x] I have tested it manually in a **supercomputer**.

Reproduce instructions:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have documented the public methods of any public class according to the guide styles. 
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes
- [x] I have rebased my branch before trying to merge.

### CODE REVIEW

File: dislib/regression/linear/base.py [hunk 1]
```diff
@@ -32,26 +40,26 @@ class LinearRegression(BaseEstimator):
     >>> import dislib as ds
     >>> from dislib.regression import LinearRegression
     >>> from pycompss.api.api import compss_wait_on
-    >>> x_data = np.array([1, 2, 3, 4, 5]).reshape(-1, 1)
+    >>> x_data = np.array([[1, 2], [2, 0], [3, 1], [4, 4], [5, 3]])
     >>> y_data = np.array([2, 1, 1, 2, 4.5]).reshape(-1, 1)
```
Comment (javicid) [inline dislib/regression/linear/base.py:44]: The reshape is no longer necessary

File: dislib/regression/linear/base.py [hunk 2]
```diff
@@ -105,88 +113,87 @@ def predict(self, x):
...
-    while len(partials) > 1:
-        partials_chunk = partials[:arity]
-        partials = partials[arity:]
-        partials.append(func(*partials_chunk))
-    return partials[0]
+    @property
+    def intercept_(self):
+        validation.check_is_fitted(self, '_intercept')
+        self._intercept = compss_wait_on(self._intercept)
+        return self._intercept
```
Comment (javicid) [inline dislib/regression/linear/base.py:132]: Wouldn't it be better to return coef and the intercept as a ds-array and let the user decide if they want to synchronize or not with collect?

File: dislib/regression/linear/base.py (hunk 1 above)
Comment (salvisolamartinell) [reply dislib/regression/linear/base.py]: Removed unnecessary reshapes

File: dislib/regression/linear/base.py (hunk 2 above)
Comment (salvisolamartinell) [reply dislib/regression/linear/base.py]: Done!

**Pull request merged**